include/saber/geometry/comparator.hpp

```diff
@@ -0,0 +1,115 @@
+#ifndef SABER_GEOMETRY_COMPARATOR_HPP
+#define SABER_GEOMETRY_COMPARATOR_HPP
+#pragma once
+
+// saber
+#include "saber/utility.hpp"
+#include "saber/geometry/point.hpp"
+#include "saber/geometry/size.hpp"
+
+// std
+#include <algorithm>
+#include <cmath>
+#include <limits>
+
+namespace saber {
+namespace geometry {
+namespace detail {
+
+bool CompareInexact(float inLHS, float inRHS)
+{
+    // magnitude: the further we get away from 0, the more inexactness we allow
+    const float magnitude = std::max(std::max(std::abs(inLHS), std::abs(inRHS)), 1.0f); 
+    // difference: compare the 2 numbers
+    const float difference = std::abs(inLHS - inRHS);
+    // epsilon: minimal permitted amount of inexactness
+    const float epsilon = std::numeric_limits<float>::epsilon() * magnitude;
+    // isEqual: equality occurs if the difference is within a scaled epsilon
+    const bool isEqual = difference < epsilon; // some arbirtary small number
+    return isEqual;
+}
+
+bool CompareInexact(double inLHS, double inRHS)
+{
+    // magnitude: the further we get away from 0, the more inexactness we allow
+    const double magnitude = std::max(std::max(std::abs(inLHS), std::abs(inRHS)), 1.0); 
+    // difference: compare the 2 numbers
+    const double difference = std::abs(inLHS - inRHS);
+    // epsilon: minimal permitted amount of inexactness
+    const double epsilon = std::numeric_limits<double>::epsilon() * magnitude;
+    // isEqual: equality occurs if the difference is within a scaled epsilon
+    const bool isEqual = difference < epsilon; // some arbirtary small number
```

Comment:
same here...
`isEqual = difference < epsilon`?
Q: less-than `<`? Or less-than-or-equal `<=`?

Consider the case of a "fully generic" algorithm.
One which also supports integral types like: `int`, `short`, or `long long`.

Q: what value is returned by `std::numeric_limits<T>::epsilon()` for integral types?
Q: what change is needed to make current algorithm also-support integral types?